core/coreapi/coreapi.go

```diff
@@ -91,6 +97,25 @@ func NewCoreAPI(n *core.IpfsNode, opts ...options.ApiOption) (coreiface.CoreAPI,
 	return (&CoreAPI{nd: n, parentOpts: *parentOpts}).WithOptions(opts...)
 }
 
+var KnownReifiers map[string]ipld2.NodeReifier = make(map[string]ipld2.NodeReifier)
+
+func (api *CoreAPI) LinkSystem() ipld2.LinkSystem {
+	lsys := cidlink.DefaultLinkSystem()
+	lsys.KnownReifiers = KnownReifiers
+	lsys.StorageReadOpener = func(linkContext linking.LinkContext, link datamodel.Link) (io.Reader, error) {
+		if cl, ok := link.(cidlink.Link); !ok {
+			return nil, fmt.Errorf("cannot process link: %v", link)
+		} else {
+			block, err := api.blocks.GetBlock(linkContext.Ctx, cl.Cid)
```

Comment:
I think there's regret around the structure we ended up here - with links in practice needing to be cidlink.Link but doing this check of it every time to pull out the Cid.

Instead, I think the pattern ipld-prime has been hoping to transition to is to use `cid.Cast(link.Binary())`

Reply:
makes sense, I thought this pattern is what we've got at the moment. Also, wouldn't `cid.Cast` do parsing again?